rust/operator-binary/src/kerberos.rs

```diff
@@ -0,0 +1,77 @@
+use snafu::{ResultExt, Snafu};
+use stackable_kafka_crd::{security::KafkaTlsSecurity, KafkaCluster, KafkaRole};
+use stackable_operator::{
+    builder::{
+        self,
+        pod::{
+            container::ContainerBuilder,
+            volume::{
+                SecretOperatorVolumeSourceBuilder, SecretOperatorVolumeSourceBuilderError,
+                VolumeBuilder,
+            },
+            PodBuilder,
+        },
+    },
+    kube::ResourceExt,
+};
+
+#[derive(Snafu, Debug)]
+pub enum Error {
+    #[snafu(display("failed to add Kerberos secret volume"))]
+    KerberosSecretVolume {
+        source: SecretOperatorVolumeSourceBuilderError,
+    },
+
+    #[snafu(display("failed to add needed volume"))]
+    AddVolume { source: builder::pod::Error },
+
+    #[snafu(display("failed to add needed volumeMount"))]
+    AddVolumeMount {
+        source: builder::pod::container::Error,
+    },
+}
+
+pub fn add_kerberos_pod_config(
+    kafka: &KafkaCluster,
+    kafka_security: &KafkaTlsSecurity,
+    role: &KafkaRole,
+    cb_kcat_prober: &mut ContainerBuilder,
+    cb_kafka: &mut ContainerBuilder,
+    pb: &mut PodBuilder,
+) -> Result<(), Error> {
+    if let Some(kerberos_secret_class) = kafka_security.kerberos_secret_class() {
+        // Mount keytab
+        let kerberos_secret_operator_volume =
+            SecretOperatorVolumeSourceBuilder::new(kerberos_secret_class)
+                .with_service_scope(kafka.name_any())
```

Comment:
Either we get the `kafka/<kafka-name>@REALM` principal working or we should be able to remove the `service` scope

Reply:
See cb8301d.